pkg/commands/crypto.go

```diff
@@ -0,0 +1,150 @@
+package commands
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io"
+	"math/big"
+	mathRand "math/rand"
+)
+
+// encrypt text using the key
+func encrypt(text, key string) ([]byte, error) {
+	gcm, err := createGCM(key)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := make([]byte, gcm.NonceSize())
```

Comment:
what benefit do we get from using a nonce here?

Reply:
I don't know, I've mostly copied over the encryption / decryption with a password code from somewhere else. I think the nonce is used as salt for the encryption. If we would not do that the encryption output would always be the same when encrypting the same string and that's would mean it's easy to decrypt.  

You'll probably already know this but GoLang's rand (random) package has a default random seed string. Because of that the output of any random function in go has the same output.

Reply:
I've got it from here: https://astaxie.gitbooks.io/build-web-application-with-golang/en/09.6.html